spicepy/_client.py

```diff
@@ -41,10 +41,10 @@ def __init__(
             tls_root_cert = Path(certifi.where())
 
         with open(tls_root_cert, 'rb') as cert_file:
```

Comment:
super nit: you're opening and reading the same cert file twice, which is not _ideal_. I would have left the open logic inside `Client` and then shared the buffer ... but this probably has minuscule performance implications, so no need to change now!

Reply:
Also, re: DI that we talked about today, this implementation is good in that the tls path is overridable, but still more prescriptive than it needs to be: all it really cares about is the tls cert _contents_, so passing it a buffer of the contents is "more DI-y" (and hence "better") than forcing it to look up these contents itself from the filesystem.

Imagine you wanted to write a test for the tls logic (eg. using a custom self-signed cert). It'd be great to do that in memory and just pass in the key here, rather than having to write to the fs and make sure this code then finds it, then tidy up after the test completes, ...

There's a limit to all of this, but sometimes it's better to push decisions _out_ of your code so that it becomes _more_ useful by doing _less_.

Again, no need to actually make this change here though 😅